yarn-project/aztec-nr/aztec/src/oracle/get_nullifier_membership_witness.nr

```diff
@@ -1,50 +1,44 @@
 use dep::protocol_types::{
+    abis::nullifier_leaf_preimage::{
+        NullifierLeafPreimage,
+        NULLIFIER_LEAF_PREIMAGE_LENGTH,
+    },
     constants::NULLIFIER_TREE_HEIGHT,
     hash::pedersen_hash,
 };
 use crate::utils::arr_copy_slice;
 
-global LEAF_DATA_LENGTH: Field = 3;
-// TODO: move this to constants.hpp so that it gets computed as INDEX_LENGTH + LEAF_DATA_LENGTH + NULLIFIER_TREE_HEIGHT
+// INDEX_LENGTH + NULLIFIER_LEAF_PREIMAGE_LENGTH + NULLIFIER_TREE_HEIGHT
 global NULLIFIER_MEMBERSHIP_WITNESS: Field = 24;
 
-// Noir version of LeafData interface from indexed merkle tree.
-// TODO(#3470) replace with /mnt/user-data/jan/aztec-packages/yarn-project/noir-protocol-circuits/src/crates/rollup-lib/src/abis/nullifier_leaf_preimage.nr
-struct LeafData {
-    value: Field,
-    next_index: Field,
-    next_value: Field,
-}
-
-impl LeafData {
-    fn serialize(self) -> [Field; LEAF_DATA_LENGTH] {
-        [self.value, self.next_index, self.next_value]
-    }
-
-    fn hash(self) -> Field {
-        // Performs the same hashing as StandardIndexedTree::encodeLeaf(...)
-        pedersen_hash(self.serialize(), 0)
-    }
-}
-
 struct NullifierMembershipWitness {
     index: Field,
-    leaf_data: LeafData,
+    leaf_preimage: NullifierLeafPreimage,
     path: [Field; NULLIFIER_TREE_HEIGHT],
 }
 
+impl NullifierMembershipWitness {
+    pub fn deserialize(fields: [Field; NULLIFIER_MEMBERSHIP_WITNESS]) -> Self {
+        let leaf_preimage_fields = arr_copy_slice(fields, [0; NULLIFIER_LEAF_PREIMAGE_LENGTH], 1);
+        Self {
+            index: fields[0],
+            leaf_preimage: NullifierLeafPreimage::deserialize(leaf_preimage_fields),
+            path: arr_copy_slice(fields, [0; NULLIFIER_TREE_HEIGHT], 1 + NULLIFIER_LEAF_PREIMAGE_LENGTH)
+        }
+    }
+}
+
 #[oracle(getLowNullifierMembershipWitness)]
-fn get_low_nullifier_membership_witness_oracle(_block_number: u32, _nullifier: Field) -> [Field; NULLIFIER_MEMBERSHIP_WITNESS] {}
+fn get_low_nullifier_membership_witness_oracle(
```

Comment:
Was the formatting not run before this?

Reply:
It was run but with an older nargo which did not wrap lines correctly. I will send a PR later in which I will fix it.